### ISSUE TITLE
Vivaldi window decorations misbehavior

### ISSUE THREAD

**Issue by Sadi58** [feren-OS/Bug-Reporting-Center#96]
**Package Name or Section of Feren OS where this bug resides**
Vivaldi

**Describe the bug**
Vivaldi shows no titlebar buttons (with Vivaldi's Settings > Appearance > Use Native Window unchecked) or continues displaying the default Feren theme titlebar buttons when user selects another theme (e.g. Breeze) via System Settings > Theme > Window Decorations.

**Steps to Reproduce**
1. Open Simple Menu and go to System Settings > Theme > Window Decorations.
2. Select Breeze.

**Expected behavior**
Vivaldi should act exactly the same as other apps and change titlebar buttons when user selects another theme for Window Decorations.

**Comment by feren** [comment]
This is because of a Window Rule in System Settings -> Other -> Window Rules to give Vivaldi shadows. To disable it, go into that area of System Settings, modify the "Certain Applications Shadows Workaround" rule and change remove 'vivaldi-stable' from the Regular Expression and save and Vivaldi will have a titlebar once more.

**Comment by feren** [comment]
Once it's done, it should have this as Window Class:
Exact Match, disabled

**Comment by Sadi58** [comment]
Thanks, but it seems there's something wrong:
1. I went to System Settings > Advanced > Other > Window Rules > (BREEZE DECORATIONS ONLY) Certain Apps Shadows Workaround > Modify > Window class (application)
2. I changed "Regular Expression: (vivaldi-stable)" to "Exact Match: disabled"
3. I logged out and back in...
No change. :-(

**Comment by feren** [comment]
You'll then want to go into System Settings -> Theme -> Window Decorations -> Feren or Breeze -> Configure button in preview -> Window-specific overrides -> Remove the vivaldi-stable rule

Forgot to mention that.

**Comment by Sadi58** [comment]
Thank you! I think I'll prefer this solution at the expense of a little compromise from the appearance of Vivaldi window - probably because I don't like Microsoft Windows style titlebar buttons.

**Comment by feren** [comment]
Vivaldi in Feren OS has a tiny CSS patch to make the titlebar buttons match with Feren OS's

**Comment by Sadi58** [comment]
Sorry to bother you again; but I can't get rid of the system titlebar on top of the Vivaldi window when I decide to rollback the settings to the original state.
There's probably a bug here (or did I break something?).
Vivaldi Settings > Appearance > Use Native Window behaves inconsistently - sometimes adding its native titlebar buttons beneath the system titlebar buttons, and sometimes not. :-(

**Comment by feren** [comment]
Just do the reverse of what you did earlier.

**Comment by Sadi58** [comment]
Thanks, I did, but apparently I had to take care of several other settings in three places:
1. KDE Plasma System Settings > Advanced > Other > Window Rules > (BREEZE DECORATIONS ONLY) Certain Apps Shadows Workaround > Modify...
2. KDE Plasma System Settings > Theme > Window Decorations > Feren/Breeze > Edit > Window-specific overrides...
3. Vivaldi Settings > Appearance > Use Native Window
At the end, I finally managed to find my way in the maze. :-)
